packages/integrations/vercel/src/lib/web-analytics.ts

```diff
@@ -0,0 +1,56 @@
+import type { AnalyticsProps } from '@vercel/analytics';
+import { fileURLToPath } from 'url';
+
+export type VercelWebAnalyticsConfig = {
+	enabled: boolean;
+	config?: Omit<AnalyticsProps, 'beforeSend'>;
+};
+
+async function getWebAnalyticsFunctions(root: URL) {
+	try {
+		const files = await Promise.all([
+			import(/* @vite-ignore */ fileURLToPath(new URL('./vercel-web-analytics.ts', root))).catch(
+				() => undefined
+			),
+			import(/* @vite-ignore */ fileURLToPath(new URL('./vercel-web-analytics.js', root))).catch(
+				() => undefined
+			),
+		]);
+
+		const functions = files[0] || files[1];
+
+		if (functions?.default) {
+			if (typeof functions.default.beforeSend !== 'function') {
+				throw new Error(
+					`@astrojs/vercel: ./vercel-web-analytics.js should export a \`beforeSend\` function.`
+				);
+			}
+
+			return {
+				beforeSend: functions.default.beforeSend,
+			};
+		}
+
+		return {
+			beforeSend: undefined,
+		};
+	} catch (e) {
+		return {
```

Comment:
I think we should still log `e`, so at least we know what went wrong during debugging. 

You can obtain the astro logger from the Astro hooks, it's called `logger`, and its type is `AstroIntegrationLogger`. 

You can pass it to `getInjectableWebAnalyticsContent`

Reply:
Amazing, didn't know that. Updated!